src/ILCompiler.Build.Tasks/src/ComputeManagedAssemblies.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+
+
+namespace Build.Tasks
+{
+    public class ComputeManagedAssemblies : Task
+    {
+        [Required]
+        public ITaskItem[] Assemblies
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The CoreRT-specific System.Private.* assemblies that must be used instead of the netcoreapp2.0 versions.
+        /// </summary>
+        [Required]
+        public ITaskItem[] SdkAssemblies
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The set of AOT-specific framework assemblies we currently need to use which will replace the same-named ones
+        /// in the app's closure.
+        /// </summary>
+        [Required]
+        public ITaskItem[] FrameworkAssemblies
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The native apphost (whose name ends up colliding with the CoreRT output binary) 
+        /// </summary>
+        [Required]
+        public string DotNetAppHostExecutableName
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The CoreCLR dotnet host fixer library that can be skipped during publish
+        /// </summary>
+        [Required]
+        public string DotNetHostFxrLibraryName
+        {
+            get;
+            set;
+        }
+
+
+        /// <summary>
+        /// The CoreCLR dotnet host policy library that can be skipped during publish
+        /// </summary>
+        [Required]
+        public string DotNetHostPolicyLibraryName
+        {
+            get;
+            set;
+        }
+
+        [Output]
+        public ITaskItem[] ManagedAssemblies
+        {
+            get;
+            set;
+        }
+
+        [Output]
+        public ITaskItem[] AssembliesToSkipPublish
+        {
+            get;
+            set;
+        }
+
+        public override bool Execute()
+        {
+            var list = new List<ITaskItem>();
+            var assembliesToSkipPublish = new List<ITaskItem>();
+
+            ITaskItem[] assemblies = this.Assemblies;
```

Comment:
Nit: This local is not useful anymore. It is used in exactly one place in the foreach.